airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -0,0 +1,172 @@
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=self.role_name)
+            self.log.info("Iam Role Name: %s", self.role_name)
+            return glue_execution_role
+        except Exception as general_error:
+            self.log.error(f'Failed to create aws glue job, error: {str(general_error)}')
+            raise
+
+    def initialize_job(self, script_arguments=None):
+        """
+        Initializes connection with AWS Glue
+        to run job
+        :return:
+        """
+        if self.s3_bucket is None:
+            raise AirflowException(
+                'Could not initialize glue job, '
+                'error: Specify Parameter `s3_bucket`'
+            )
+
+        glue_client = self.get_conn()
+
+        try:
+            job_name = self.get_or_create_glue_job()
+            job_run = glue_client.start_job_run(
+                JobName=job_name,
+                Arguments=script_arguments
+            )
+            return self.job_completion(job_name, job_run['JobRunId'])
+        except Exception as general_error:
+            self.log.error(f'Failed to run aws glue job, error: {str(general_error)}')
+            raise
+
+    def job_completion(self, job_name=None, run_id=None):
+        """
+        :param job_name: unique job name per AWS account
+        :type job_name: str
+        :param run_id: The job-run ID of the predecessor job run
+        :type run_id: str
+        :return:
+        """
+        while True:
+            glue_client = self.get_conn()
+            job_status = glue_client.get_job_run(
+                JobName=job_name,
+                RunId=run_id,
+                PredecessorsIncluded=True
+            )
+            job_run_state = job_status['JobRun']['JobRunState']
+            done_states = ['FAILED', 'STOPPED', 'SUCCEEDED']
+            if job_run_state in done_states:
+                self.log.info("Exiting Job %s Run State: %s", run_id, job_run_state)
+                return {'JobRunState': job_run_state, 'JobRunId': run_id}
+            else:
+                self.log.info("Polling for AWS Glue Job %s current run state with status %s"
+                    , job_name, job_run_state)
+                time.sleep(self.JOB_POLL_INTERVAL)
+
+    def get_or_create_glue_job(self):
+        """
+        Creates(or just returns) and returns the Job name
+        :return:Name of the Job
+        """
+        glue_client = self.get_conn()
+        try:
+            get_job_response = glue_client.get_job(JobName=self.job_name)
+            self.log.info("Job Already exist. Returning Name of the job")
+            return get_job_response['Job']['Name']        
+        except glue_client.exceptions.EntityNotFoundException:
+            self.log.info("Job doesnt exist. Now creating and running AWS Glue Job")
+            s3_log_path = f's3://{str(self.s3_bucket)}/{str(self.S3_GLUE_LOGS)}{str(self.job_name)}'
```

Comment:
```suggestion
            s3_log_path = f's3://{self.s3_bucket}/{self.S3_GLUE_LOGS}{self.job_name}'
```
I don't think you need to convert it to `str` before. All these value are strings already.